Assets/MixedRealityToolkit/Providers/UnityInput/MouseController.cs

```diff
@@ -145,7 +138,14 @@ public void Update()
 
                 if (Interactions[i].InputType == DeviceInputType.Scroll)
                 {
-                    Interactions[i].Vector2Data = UInput.mouseScrollDelta;
+                    Vector2 wheelDelta = UInput.mouseScrollDelta;
+                    if (mouseDeviceManager != null)
```

Comment:
>mouseDeviceManager [](start = 24, length = 18)

Something I don't quite understand - if device managers are creating controllers, how is that this could be null?

Reply:
I see some other null checks sprinkled around the code, and I know it's consistent with that, but I don't really understand why that's necessary in the first place.

---
In reply to: [321811241](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5890#discussion_r321811241) [](ancestors = 321811241)

Reply:
Since we do not pass the device manager instance to the controller object, we need to fetch it from the input system. If the input system does not support the IMixedRealityDataProviderAccess interface, the controller will not be able to get the device manager.

This should not happen with the MRTK provided input system, but may with a custom implementation.